### DISCUSSION TITLE
how to get attached images from a wordpress import to be included ?

### DISCUSSION THREAD

**Discussion by netllama** [getpelican/pelican#3049] [Q&A]
Greetings!
I've got pelican-4.5.4 installed on a Fedora Linux system.

I've got several Wordpress blogs which have made heavy use of attached images in their posts (some of them are dedicated photography blogs where the entire post is literally just a single attached image with no other text).  I'd like to import each of these blogs into Pelican such that the generated "output" for the posts automatically includes the attached image(s), but I'm not quite understanding how to make it work.  I thought the --wp-attach option would make the magic happen:

    pelican-import --wpfile --wp-attach ~/wp-export.xml

It did download all of the images (great), but the resulting generated HTML makes no reference to the images, and all of the posts are basically empty when viewed in a web browser.  The help for the --wp-attach option says "Files will be added to posts as a list in the post header and links to the files within the post will be updated." but I don't think that's actually happening. I see all of the downloaded images sitting under the content directory (content/year6/wp-content/uploads/...) but none of them made it over to the output directory.

What am I missing here?  What did I do wrong?

thanks!

**Comment by avaris** [comment]
Hard to know without further information. It may be the case that images are placed into the metadata only, if you have attached images but there is no reference to the image within the article (in the wordpress export). Is it possible to share a relevant article, i.e. the entry within the XML export and the resulting content (MD or RST) for that article?

**Comment by netllama** [comment]
Sure, here's the XML for the export of one article:
```
<item>
                <title><![CDATA[2015-12-15]]></title>
                <link>https://netllama.linux-sxs.org/year6/?p=1174</link>
                <pubDate>Tue, 15 Dec 2015 20:58:35 +0000</pubDate>
                <dc:creator><![CDATA[netllama]]></dc:creator>
                <guid isPermaLink="false">http://netllama.linux-sxs.org/year6/?p=1174</guid>
                <description></description>
                <content:encoded><![CDATA[]]></content:encoded>
                <excerpt:encoded><![CDATA[]]></excerpt:encoded>
                <wp:post_id>1174</wp:post_id>
                <wp:post_date><![CDATA[2015-12-15 12:58:35]]></wp:post_date>
                <wp:post_date_gmt><![CDATA[2015-12-15 20:58:35]]></wp:post_date_gmt>
                <wp:post_modified><![CDATA[2016-01-03 12:59:11]]></wp:post_modified>
                <wp:post_modified_gmt><![CDATA[2016-01-03 20:59:11]]></wp:post_modified_gmt>
                <wp:comment_status><![CDATA[open]]></wp:comment_status>
                <wp:ping_status><![CDATA[closed]]></wp:ping_status>
                <wp:post_name><![CDATA[2015-12-15]]></wp:post_name>
                <wp:status><![CDATA[publish]]></wp:status>
                <wp:post_parent>0</wp:post_parent>
                <wp:menu_order>0</wp:menu_order>
                <wp:post_type><![CDATA[post]]></wp:post_type>
                <wp:post_password><![CDATA[]]></wp:post_password>
                <wp:is_sticky>0</wp:is_sticky>
                                                                                                                <wp:postmeta>
                <wp:meta_key><![CDATA[_wpt_status_message]]></wp:meta_key>
                <wp:meta_value><![CDATA[Tweet sent successfully.]]></wp:meta_value>
                </wp:postmeta>
                                                        <wp:postmeta>
                <wp:meta_key><![CDATA[_jd_wp_twitter]]></wp:meta_key>
                <wp:meta_value><![CDATA[a:1:{i:0;s:47:"New Year6 post: 2015-12-15 http://goo.gl/ShztEr";}]]></wp:meta_value>
                </wp:postmeta>
                                                        <wp:postmeta>
                <wp:meta_key><![CDATA[_wp_jd_target]]></wp:meta_key>
                <wp:meta_value><![CDATA[http://netllama.linux-sxs.org/year6/?p=1174]]></wp:meta_value>
                </wp:postmeta>
                                                        <wp:postmeta>
                <wp:meta_key><![CDATA[_wpt_short_url]]></wp:meta_key>
                <wp:meta_value><![CDATA[http://goo.gl/ShztEr]]></wp:meta_value>
                </wp:postmeta>
                                                        <wp:postmeta>
                <wp:meta_key><![CDATA[_jd_tweet_this]]></wp:meta_key>
                <wp:meta_value><![CDATA[yes]]></wp:meta_value>
                </wp:postmeta>
                                                        <wp:postmeta>
                <wp:meta_key><![CDATA[_jd_twitter]]></wp:meta_key>
                <wp:meta_value><![CDATA[]]></wp:meta_value>
                </wp:postmeta>
                                                        <wp:postmeta>
                <wp:meta_key><![CDATA[_thumbnail_id]]></wp:meta_key>
                <wp:meta_value><![CDATA[948]]></wp:meta_value>
                </wp:postmeta>
                                                        <wp:postmeta>
                <wp:meta_key><![CDATA[_edit_last]]></wp:meta_key>
                <wp:meta_value><![CDATA[1]]></wp:meta_value>
                </wp:postmeta>
                                                        </item>
```

and the markdown that pelican generated for that same article:
```
Title: 2015-12-15
Date: 2015-12-15 12:58
Author: netllama
Slug: 2015-12-15
Status: published
```

**Reply by avaris** [reply]
Well, there is no content in that item, so no surprise it ended up empty. Also only (vague) reference to an image I can see is:

```
            <wp:postmeta>
                <wp:meta_key><![CDATA[_thumbnail_id]]></wp:meta_key>
                <wp:meta_value><![CDATA[948]]></wp:meta_value>
            </wp:postmeta>
```

I guess `948` is referring to some image, hopefully also identified in the export XML. But currently `pelican-import` does not know how to handle this.

**Comment by netllama** [comment]
Right, this was from a photography blog where the only content in a post
was an image.

**Reply by avaris** [reply]
Yeah, that's a different way of adding images and looks like `pelican-import` could use an update but that requires some example XML export: i.e. to identify which image `948` refers to and insert it into the content.

**Comment by netllama** [comment]
I'm happy to provide the full XML export if that would be helpful?
Its about 1.7MB in size.

**Reply by avaris** [reply]
Someone needs to implement it :). I can't promise I'll have any time working on it but sure if it is OK to share it, XML will be very helpful.

**Comment by netllama** [comment]
[year6.WordPress.2022-09-18.xml.gz](https://github.com/getpelican/pelican/files/9815207/year6.WordPress.2022-09-18.xml.gz)

OK, I'm attaching a gzipped version of the full XML.

**Reply by avaris** [reply]
Thanks! I think I understand how this works. `_thumbnail_id` refers to the `post_id` of the `item` of type `attachment`. Need to associate items properly and possibly insert a link to the image in the content.
